### PR TITLE
feat: Keyman Settings Manager base classes

### PR DESCRIPTION
**Pull request by mcdurdin**
Introduces a new settings system that lists all available debug settings in Keyman Desktop and Developer with a single unified UI to manage them (separate PR #3733) as well as a console app (#3732). This comes because I (as lead dev) am always forgetting what the various keys and values are and need a better way to manage them. 

This PR introduces the Keyman Settings Manager as a global shared class.

Note: this PR does not include the standard Keyman options which can be configured via the normal Keyman Configuration Options dialog. I don't anticipate adding these standard options in Keyman 14; perhaps in 15.0.

### CODE REVIEW

File: windows/src/global/delphi/general/Keyman.System.Settings.pas [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+                   'pipeline/ for details';
+      DefaultInt: 1;
+      ValueType: kstInteger
+    ),
+
+    //
+    // engine.diagnostics
+    //
+
+    // keyman32:k32_globals.Globals::LoadDebugSettings
```
Comment (mcdurdin) [inline windows/src/global/delphi/general/Keyman.System.Settings.pas:416]: You'll note that this is the only setting which is also available in the Keyman Options.

File: windows/src/global/delphi/general/Keyman.System.SettingsManagerFile.pas [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  ID: string;
+  s: TKeymanSetting;
+  json: TJSONObject;
+  stream: TStringStream;
+  p: TJSONPair;
+begin
+  stream := TStringStream.Create('', TEncoding.UTF8);
+  try
+    stream.LoadFromFile(Filename);
+    json := TJSONObject.ParseJSONValue(stream.DataString, False, True) as TJSONObject;
```
Comment (darcywong00) [inline windows/src/global/delphi/general/Keyman.System.SettingsManagerFile.pas:39]: Are there any warnings if importing invalid JSON?

**Review by darcywong00 — APPROVED** [review]
lgtm

File: windows/src/global/delphi/general/Keyman.System.SettingsManagerFile.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/global/delphi/general/Keyman.System.SettingsManagerFile.pas:39]: Invalid JSON causes an exception which will be handled by the caller

**Pull request merged**